src/completions/generate-v2.ts

```diff
@@ -0,0 +1,241 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { core } from '../updates/appc';
+
+import { CompletionsFormat } from './index';
+import { CustomError, getAlloyCompletionsFileName, getAlloyVersion, getSDKCompletionsFileName } from './util';
+
+import os from 'os';
+
+interface TagDictionary {
+	[key: string]: Tag;
+}
+interface Tag {
+	apiName: string;
+}
+
+interface PropertiesDictionary {
+	[key: string]: Property;
+}
+interface Property {
+	description: string;
+	type: string;
+	values?: string[];
+}
+
+interface TypeDictionary {
+	[key: string]: Type;
+}
+interface Type {
+	description: string;
+	events: string[];
+	functions: string[];
+	properties: string[];
+}
+
+/**
+ * Generate completions for an Alloy version.
+ *
+ * @param {Boolean} force - Force generation of completion file.
+ */
+export async function generateAlloyCompletions (force: boolean) {
+	const appcPath = path.join(os.homedir(), '.appcelerator', 'install');
+	const version = await core.checkInstalledVersion();
+	if (!version) {
+		throw Error('Unable to find installed alloy version.');
+	}
+	const alloyPath = path.join(appcPath, version, 'package', 'node_modules', 'alloy');
+	const alloyVersion = await getAlloyVersion();
+
+	const alloyCompletionsFilename = getAlloyCompletionsFileName(alloyVersion, CompletionsFormat.v2);
+
+	if (!force && await fs.pathExists(alloyCompletionsFilename)) {
+		return;
+	}
+
+	const alloyAPIPath = path.join(alloyPath, 'docs', 'api.jsca');
+	const api = await fs.readJSON(alloyAPIPath);
+
+	const { props, types } = await getValues(api);
```

Comment:
I'm not really sure what `getValues` does here? Could we make the name more descriptive?